framework/src/transfers/MultiAppFieldTransferInterface.C

```diff
@@ -0,0 +1,102 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "MultiAppFieldTransferInterface.h"
+#include "MooseTypes.h"
+#include "FEProblem.h"
+#include "MultiApp.h"
+#include "MooseMesh.h"
+
+template <>
+InputParameters
+validParams<MultiAppFieldTransferInterface>()
+{
+  InputParameters params = validParams<MultiAppTransfer>();
+  params.addRequiredParam<AuxVariableName>(
```

Comment:
Is the restriction on aux var necessary? Looks like the code can work with nonlinear variables.

Reply:
Transfers could happen in anytime (nonlinear, linear). The solution, in theory, is locked by PETSc, and we should not change that vector anyway (you definitely  can if you really want). 

The solution vector in the auxiliary system is accessible since it is not touched by solvers.

I could be totally wrong on this, and there are other reasons @friedmud  and @permcody may be aware of.